pageserver/src/http/routes.rs

```diff
@@ -77,6 +77,15 @@ fn check_permission(request: &Request<Body>, tenant_id: Option<TenantId>) -> Res
     })
 }
 
+fn apierror_from_prerror(err: PageReconstructError) -> ApiError {
```

Comment:
NIT: with a read `impl From<PageReconstructError> for `ApiError`, supposedly `?` would work everywhere without explicit `map_err`.

Reply:
I'm against magic `From` impls unless they're completely mechanical.

This one isn't because of the `wal redo failed` prefixing that this function does.

Reply:
Reverted the `From` impl